File: Bai_118/main.py
n = int(input())
a = [0] * n
f = [0] * (n + 1)
for i in range(1, n):
    a[i]= int(input())
f[2] = a[1]
f[3] = a[1] + a[2]
for i in range(4, n + 1):
    f[i] = min(f[i - 2] + a[i - 1], f[i - 1] + a[i - 1])
print(f[n])

File: Bai_118/main.cpp
#include<bits/stdc++.h>
using namespace std;

using ll = long long;

int main(){
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    int n; cin >> n;
    int a[n], f[n + 1] = {0}; 
    for(int i = 1; i < n; ++i) cin >> a[i];
    f[2] = a[1], f[3] = a[1] + a[2];
    for(int i = 4; i <= n; ++i)
        f[i] = min(f[i - 2] + a[i - 1], f[i - 1] + a[i - 1]);
    cout << f[n];
    return 0;
}